ode-wrapper/src/main/java/com/trihydro/odewrapper/helpers/SetItisCodes.java

```diff
@@ -163,24 +162,56 @@ public List<String> setItisCodes(WydotTim wydotTim) {
                 items.add(code.getItisCode().toString());
         }
 
-    return items;
-  }
+        return items;
+    }
+
+    public List<String> setItisCodesVsl(WydotTimVsl wydotTim) {
+
+        List<String> items = new ArrayList<String>();
+
+        List<ItisCode> allItisCodes = getItisCodes();
+
+        // add advisory code first if advisory speed limit
+        wydotTim.getItisCodes().stream().filter(x -> x.contains("7712")).findFirst()
+                .ifPresent(items::add);
+
+        // speed limit itis code
+        allItisCodes.stream().filter(x -> x.getDescription().equals("speed limit")).findFirst().ifPresent(mph -> items.add(mph.getItisCode().toString()));
 
-  public List<String> setItisCodesBowr(WydotTimBowr tim) throws WeightNotSupportedException {
-    List<String> itisCodes = new ArrayList<>();
+        // add reduced code next if reduced speed
+        wydotTim.getItisCodes().stream().filter(x -> x.contains("12302")).findFirst()
+                .ifPresent(items::add);

```

Comment:
Question (non-blocking): What does "12302" correspond to? Is that "reduced" or "reduced speed ahead"?